modules/issue_tracker/jsx/editIssue.js

```diff
@@ -0,0 +1,628 @@
+                        }
+
+                        var formData = {
+                            'issueID': data.issueData.issueID,
+                            'title': data.issueData.title,
+                            'lastUpdate': data.issueData.lastUpdate,
+                            'centerID': data.issueData.centerID,
+                            'PSCID': data.issueData.PSCID,
+                            'reporter': data.issueData.reporter,
+                            'assignee': data.issueData.assignee,
+                            'status': data.issueData.status,
+                            'priority': data.issueData.priority,
+                            'watching': data.issueData.watching,
+                            'visitLabel': data.issueData.visitLabel,
+                            'dateCreated': data.issueData.dateCreated,
+                            'category': data.issueData.category,
+                            'lastUpdatedBy': data.issueData.lastUpdatedBy,
+                            'commentHistory': data.issueData.commentHistory,
+                            'comment': data.issueData.comment,
+                            'otherWatchers': data.issueData.whoIsWatching
+                        };
+
+                        console.log(data.otherWatchers);
+
+                        that.setState(
+                            {
+                                'Data': data,
+                                'isLoaded': true,
+                                'issueData': data.issueData,
+                                'formData': formData
+                            }
+                        );
+
+                        that.setState(
+                            {
+                                'error': "finished success"
+                            }
+                        )
+                    },
+                    error: function (data, error_code, error_msg) {
+                        that.setState(
+                            {
+                                'error': error_msg
+                            }
+                        );
+                    }
+                }
+            );
+        },
+
+
+        /**
+         * Handles form submission
+         *
+         * @param e
+         */
+        handleSubmit: function (e) {
+            e.preventDefault();
+
+            var self = this;
+            var myFormData = this.state.formData;
+            var formRefs = this.refs;
+            var formData = new FormData();
+            var issueData = this.state.issueData;
+
+            // Validate the form
+            if (!this.isValidForm(formRefs, myFormData)) {
+                return;
+            }
+
+            for (var key in myFormData) {
+                if (myFormData[key] != "" && myFormData[key] !== issueData[key]) {
+                    formData.append(key, myFormData[key]);
+                }
+            }
+
+            formData.append('issueID', myFormData['issueID']);
```

Comment:
You seem to have 2 state vars that track issueID, the one that is sent to the server is in this.state.formData.issueID and it seems to be always null. If you use this.state.issueID instead of `myFormData['issueID']` it should work.